lib/src/main/java/graphql/nadel/NadelSchemas.kt

```diff
@@ -52,45 +54,69 @@ data class NadelSchemas constructor(
         }
 
         fun overallSchema(serviceName: String, schema: Reader): Builder = also {
-            overallSchemas[serviceName] = schema
+            overallSchemaReaders[serviceName] = schema
         }
 
         fun underlyingSchema(serviceName: String, schema: Reader): Builder = also {
-            underlyingSchemas[serviceName] = schema
+            underlyingTypeDefs.remove(serviceName) // remove from other Map
+            underlyingSchemaReaders[serviceName] = schema
+        }
+
+        fun underlyingSchema(serviceName: String, schema: TypeDefinitionRegistry): Builder = also {
+            underlyingSchemaReaders.remove(serviceName) // remove from other Map
+            underlyingTypeDefs[serviceName] = schema
         }
 
         fun overallSchema(serviceName: String, schema: String): Builder = also {
-            overallSchemas[serviceName] = schema.reader()
+            overallSchemaReaders[serviceName] = schema.reader()
         }
 
         fun underlyingSchema(serviceName: String, schema: String): Builder = also {
-            underlyingSchemas[serviceName] = schema.reader()
+            underlyingSchema(
+                serviceName,
+                MultiSourceReader.newMultiSourceReader()
```

Comment:
Need to use GraphQL Java's `MultiSourceReader` otherwise you don't get line numbers… TIL